extensions/amazon-aws/src/actions/logs.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  CloudWatchLogsClient,
+  DescribeLogStreamsCommand,
+  FilteredLogEvent,
+  FilterLogEventsCommand,
+  LogStream,
+} from "@aws-sdk/client-cloudwatch-logs";
+import { AWS_URL_BASE } from "../constants";
+import { LogStartTimes } from "../interfaces";
+
+const cloudWatchLogsClient = new CloudWatchLogsClient({});
+export async function fetchLogs(
+  logGroupName: string,
+  startTime: LogStartTimes,
+  logStreamNamePrefix?: string,
+  logStreamNames?: string[]
+): Promise<FilteredLogEvent[]> {
+  if (!process.env.AWS_PROFILE) return [];
+  return await fetchAllLogs(logGroupName, startTime, logStreamNamePrefix, logStreamNames);
+}
+
+export async function fetchLogStreams(logGroupName: string): Promise<LogStream[]> {
+  if (!process.env.AWS_PROFILE) return [];
+  return await fetchAllLogStreams(logGroupName);
+}
+
+async function fetchAllLogStreams(logGroupName: string, token?: string, accEvents?: LogStream[]): Promise<LogStream[]> {
+  const { logStreams, nextToken } = await cloudWatchLogsClient.send(
+    new DescribeLogStreamsCommand({ logGroupName, nextToken: token })
+  );
+
+  const combinedEvents = [...(accEvents || []), ...(logStreams || [])];
+
+  if (nextToken) {
+    return fetchAllLogStreams(logGroupName, nextToken, combinedEvents);
+  }
+
+  return combinedEvents;
+}
```

Comment:
These functions could be combined. In the current setup `fetchLogStreams` is not really doing anything meaningful.